pkg/operator/startupmonitorreadiness/readiness_checks.go

```diff
@@ -248,9 +266,13 @@ func checkRevision(kasPod *corev1.Pod, monitorRevision int) (bool, string, strin
 //	returns true, "", "", when we got HTTP 200 "successThreshold" times
 //	returns false, "NotReady", EntireResponseBody (if any) on HTTP != 200
 //	returns false, "NotReadyError", EntireResponseBody (if any) in case of any error or timeout
-func goodReadyzEndpoint(client *http.Client, rawURL string, successThreshold int, interval time.Duration) func(ctx context.Context) (bool, string, string) {
+func goodReadyzEndpoint(client *http.Client, rawURL string, successThreshold int, interval time.Duration, isSNO *bool) func(ctx context.Context) (bool, string, string) {
+	readyzURL := fmt.Sprintf("%s/readyz?verbose=true", rawURL)
+	if *isSNO {
+		readyzURL = fmt.Sprintf("%s&exclude=api-openshift-apiserver-available", readyzURL)
```

Comment:
Is the behaviour of the `exclude` query param documented somewhere ?

Reply:
it looks like it is - https://kubernetes.io/docs/reference/using-api/health-checks/

Reply:
where is `api-openshift-apiserver-available` defined ? 
what will we do if the name ever changes ? do we care ?

Reply:
> where is api-openshift-apiserver-available defined ?

okay, it looks like I have found it here:
https://github.com/openshift/kubernetes/blob/53f3066f47d3a594d6b93843916e8711ee673715/openshift-kube-apiserver/openshiftkubeapiserver/sdn_readyz_wait.go#L21

Reply:
adding a test for checking the name of the check might be very expensive.

maybe at least we should add a comment (sdn_readyz_wait.go) saying that the name should be stable otherwise this check will fail again?

Reply:
it looks like the readyz EP prints a warning if the excluded check doesn't match - i hope we do log the output from the EP somewhere.

Reply:
>maybe at least we should add a comment (sdn_readyz_wait.go) saying that the name should be stable otherwise this check will fail again?

iiuc its assumed that check names are persistent. Not sure if its part of the enhancement to establish that - I can dig it. In any case, the check won't fail if we specify an exclude which doesn't match anything

Reply:
>it looks like the readyz EP prints a warning if the excluded check doesn't match - i hope we do log the output from the EP somewhere.

kube-apiserver prints the list of checks which were run and passed/failed - along with the excluded ones (`[ ] openshift-apiserver - excluded`)

Reply:
> iiuc its assumed that check names are persistent. Not sure if its part of the enhancement to establish that 

I don't think there was an enhancement for the openshift readiness check, it was added ad-hoc.

I'm fine with either adding a test that verifies that the check we are interested in shows up on the list or with adding a comment with some instructions what to do in case the name of the check changes.

Reply:
Looks like we don't have a unit/e2e test checking kube-apiserver returned status, so I added instructions on how to update the exclude list in the commit message